sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -848,6 +850,37 @@ def test_early_stopping_on_test_set_with_warm_start():
     gb.fit(X, y)
 
 
+def test_early_stopping_with_sample_weights(monkeypatch):
+    """Check that sample weights is passed in to scorer."""
+
+    mock_scorer = Mock(side_effect=get_scorer("neg_median_absolute_error"))
+
+    def mock_check_scoring(estimator, scoring):
+        assert scoring == "neg_median_absolute_error"
+        return mock_scorer
+
+    monkeypatch.setattr(
+        sklearn.ensemble._hist_gradient_boosting.gradient_boosting,
+        "check_scoring",
+        mock_check_scoring,
+    )
+
+    X, y = make_regression(random_state=0)
+    sample_weight = np.ones_like(y)
+    hist = HistGradientBoostingRegressor(
+        max_iter=2,
+        early_stopping=True,
+        random_state=0,
+        scoring="neg_median_absolute_error",
+    )
+    hist.fit(X, y, sample_weight=sample_weight)
+
+    # For scorer is called three times per iteration. (2 x 3 = 6)
```

Comment:
I am trying to investigate by I don't understand this. I would have expected 3 in total: one for the baseline score call at line 614 and 2 for the line score at the end of each iteration at line 789.

There might be a bug.

Reply:
Actually this test should also check the contents of `hist.train_score_` and `hist.validation_score_`. The values are wrong at the moment:

```python
[make_scorer(median_absolute_error, greater_is_better=False)
 make_scorer(median_absolute_error, greater_is_better=False)
 make_scorer(median_absolute_error, greater_is_better=False)]
[make_scorer(median_absolute_error, greater_is_better=False)
 make_scorer(median_absolute_error, greater_is_better=False)
 make_scorer(median_absolute_error, greater_is_better=False)]
```